python/pyiceberg/avro/decoder.py

```diff
@@ -115,7 +115,8 @@ def read_bytes(self) -> bytes:
         """
         Bytes are encoded as a long followed by that many bytes of data.
         """
-        return self.read(self.read_int())
+        num_bytes = self.read_int()
+        return self.read(num_bytes) if num_bytes > 0 else b""
```

Comment:
Ah, this makes more sense! So we were requesting 0 bytes and hitting an `EOFError` because of it.

Reply:
It would also make sense to put this behavior into `read(num_bytes)` since we could have other callers ask for 0 bytes by accident.